components/link_tooltip/link_tooltip.tsx

```diff
@@ -51,40 +51,38 @@ export default class LinkTooltip extends React.PureComponent<Props, State> {
     public showTooltip = (e: React.MouseEvent<HTMLSpanElement>): void => {
         //clear the hideTimeout in the case when the cursor is moved from a tooltipContainer child to the link
         window.clearTimeout(this.hideTimeout);
-
-        if (!this.state.show) {
-            const target = e.currentTarget;
-            const tooltipContainer = this.tooltipContainerRef.current;
-
-            //clear the old this.showTimeout if there is any before overriding
-            window.clearTimeout(this.showTimeout);
-
-            this.showTimeout = window.setTimeout(() => {
-                this.setState({show: true});
-
-                if (!tooltipContainer) {
-                    return;
-                }
-
-                const addChildEventListeners = (node: Node) => {
-                    node.addEventListener('mouseover', () => clearTimeout(this.hideTimeout));
-                    (node as HTMLElement).addEventListener('mouseleave', (event) => {
-                        if (event.relatedTarget !== null) {
-                            this.hideTooltip();
-                        }
-                    });
-                };
-                tooltipContainer.childNodes.forEach(addChildEventListeners);
-
-                this.popper = new Popper(target, tooltipContainer, {
-                    placement: 'bottom',
-                    modifiers: {
-                        preventOverflow: {enabled: false},
-                        hide: {enabled: false},
-                    },
+        let tooltipContainer: Element | null;
+        this.setState({show: true}, () => {
+            tooltipContainer = this.tooltipContainerRef.current;
+        });
```

Comment:
What are the implications of this change of logic? It seems that we're no longer short-circuiting this function using `this.state.show` as a guard.

Will the below solution cause any issues in regards to use `this.state.show`?

```suggestion
        if (this.state.show) {
            return;
        }

        let tooltipContainer: Element | null;
        this.setState({show: true}, () => {
            tooltipContainer = this.tooltipContainerRef.current;
        });
```

Reply:
Thanks for the comment. 

I couldn't think any implications for not having `this.state.show` as a guard. 
Because, in the constructor, `this.state.show` is set to `false`. Then 
- on `mouseover`, it is set to `true` 
- on `mouseleave`, it is set back to `false` (the loop is closed and the tooltip is hidden)

In order to be 100% secure, I added your suggestion as a guard, just in case any edge case that I couldn't envisaged. 
Additionally, your suggested solution doesn't cause any issues regarding to use `this.state.show`, so we are safe here.

Please let me know if I am missing sth